src/error.rs

```diff
@@ -44,49 +38,13 @@ pub enum ClamdError {
     /// is somehow malformed. Contains the invalid response.
     #[error("incomplete response from clamd: {0}")]
     IncompleteResponse(String),
-    /// Occurs when everything between this library and clamd went
-    /// well but clamd seems to have found a virus signature. See also
-    /// [`ClamdError::scan_error`].
-    #[error("clamd returned error on scan, possible virus: {0}")]
-    ScanError(String),
-}
+    /// Occurs when communication went well with clamd but it responded
+    /// with `ERROR`
+    #[error("clamd returned error on scan of {0}: {1}")]
+    ScanError(String, String),
 
-impl ClamdError {
-    /// If you want to ignore any error but an actual malignent scan
-    /// result from clamd. I do not recommend using this without careful thought, as any other error
-    /// could hide that uploaded bytes are actually a virus.
-    /// # Example
-    /// ```rust
-    /// # use std::net::SocketAddr;
-    /// # use clamd_client::{ClamdClientBuilder, ScanResult};
-    /// # use eyre::Result;
-    /// # async fn doc() -> eyre::Result<()> {
-    /// let address = "127.0.0.1:3310".parse::<SocketAddr>()?;
-    /// let mut clamd_client = ClamdClientBuilder::tcp_socket(address)?.build();
-    ///
-    /// // This downloads a virus signature that is benign but trips clamd.
-    /// let eicar_bytes = reqwest::get("https://secure.eicar.org/eicarcom2.zip")
-    ///   .await?
-    ///   .bytes()
-    ///   .await?;
-    ///
-    /// let result = clamd_client.scan_bytes(&eicar_bytes).await?;
-    /// match result {
-    ///     ScanResult::Malignent { infection_types } => {
-    ///         tracing::info!("clamd found a virus(es):\n{}", infection_types.join("\n"))
-    ///     }
-    ///     _ => (),
-    /// };
-    /// # Ok(())
-    /// # }
-    /// ```
-    pub fn scan_error(self) -> Option<String> {
-        match self {
-            Self::ScanError(s) => Some(s),
-            _ => {
-                trace!("ignoring non-scan error {:?}", self);
-                None
-            }
-        }
-    }
+    /// Occurs when the response from clamd is not what the library
+    /// expects. Contains the invalid response.
+    #[error("invalid response from clamd: {0}")]
+    InvalidResponse(String),
```

Comment:
Why was this block moved? It's triggering my OCD (it's not in alphabetical order anymore 😄)

Reply:
Well, it wasn't in the first place anyways ha